python_modules/airline-demo/airline_demo/solids.py

```diff
@@ -297,69 +291,83 @@ def upload_to_s3(context, file_path):
         )
     ),
     description='Extracts an archive member from a zip archive.',
-    outputs=[
-        OutputDefinition(
-            List(FileExistsAtPath), description='The path to the unzipped archive member.'
-        )
-    ],
+    outputs=[OutputDefinition(List(Path), description='The path to the unzipped archive member.')],
 )
-def unzip_file(
-    context,
-    archive_paths,
-    archive_members,
-    # destination_dir=None
-):
-    # FIXME
-    # archive_path = info.config['archive_path']
-    # archive_member = info.config['archive_member']
+def unzip_file(context, archive_paths, archive_members):
+    def extract(context, archive_path, zip_ref, archive_member, destination_dir, skip_if_present):
+        filesystem = context.resources.filesystem
+
+        target_path = filesystem.join(destination_dir, archive_member)
+        is_file = filesystem.is_file(target_path)
+        is_dir = filesystem.is_dir(target_path)
+
+        if not (skip_if_present and (is_file or is_dir)):
+            tempdir = tempfile.mkdtemp()
+            try:
+                extracted_path = zip_ref.extract(archive_member, tempdir)
+                with filesystem.write(
+                    filesystem.join(destination_dir, os.path.basename(extracted_path))
+                ) as write_obj:
+                    with open(extracted_path, 'rb') as read_obj:
```

Comment:
the double with block is a little confusing here actually. I think it might be helpful to hide this behind a function